augur/curate/apply_geolocation_rules.py

```diff
@@ -98,7 +100,12 @@ def get_annotated_geolocation(geolocation_rules, raw_geolocation, case_sensitive
     # Traverse the geolocation rules based using the rule_traversal values
     for field_value in rule_traversal:
         # Use lowercase for field_value for case-insensitive rule matching
-        current_rules = current_rules.get(field_value.lower())
+        try:
+            lower_field_value = field_value.lower()
+        except AttributeError:
+            print_err(f"WARNING: Could not convert {field_value!r} to lowercase.")
+            continue
+        current_rules = current_rules.get(lower_field_value)
```

Comment:
No need for try/except – the case of `field_value` not being a string should be handled elsewhere (done in this PR).